core/src/main/java/org/bitcoinj/wallet/WalletEx.java

```diff
@@ -618,23 +622,28 @@ public List<CompactTallyItem> selectCoinsGroupedByAddresses(boolean skipDenomina
     }
 
 
+    /**
+     * Count the number of unspent outputs that have a certain value
+     */
     public int countInputsWithAmount(Coin inputValue) {
         int count = 0;
         for (Transaction tx : getTransactionPool(WalletTransaction.Pool.UNSPENT).values()) {
             for (TransactionOutput output : tx.getOutputs()) {
-                if (output.getValue().equals(inputValue) && output.isMine(this))
+                if (output.getValue().equals(inputValue) && output.isMine(this) && output.getSpentBy() == null)
```

Comment:
This fixes a minor bug where spent outputs were counted.  Over time this would result in some denominations from being unused, especially 0.001 and 0.01.